enterprise/server/hostedrunner/hostedrunner.go

```diff
@@ -220,16 +223,62 @@ func (r *runnerService) withCredentials(ctx context.Context, req *rnpb.RunReques
 		return nil, err
 	}
 
+	// If no access token is provided explicitly, try fetching the token.
+	accessToken := req.GetGitRepo().GetAccessToken()
+	if accessToken == "" {
+		repoURL, err := git.NormalizeRepoURL(req.GetGitRepo().GetRepoUrl())
+		if err != nil {
+			return nil, status.WrapError(err, "normalize git repo url")
+		}
+		isGitRepo := repoURL.Hostname() == "github.com"
+
+		if isGitRepo {
+			gitToken, err := r.getGitToken(ctx, u.GetGroupID(), repoURL.String())
+			if err != nil {
+				log.Warningf("Could not fetch git auth token for %s for hosted runner"+
+					" (Note: The token is not needed for public repos): %s", repoURL, err)
+			}
+			accessToken = gitToken
+		}
+
+	}
+
 	// Use env override headers for credentials.
 	envOverrides := []*repb.Command_EnvironmentVariable{
 		{Name: "BUILDBUDDY_API_KEY", Value: apiKey.Value},
 		{Name: "REPO_USER", Value: req.GetGitRepo().GetUsername()},
-		{Name: "REPO_TOKEN", Value: req.GetGitRepo().GetAccessToken()},
+		{Name: "REPO_TOKEN", Value: accessToken},
 	}
 	ctx = withEnvOverrides(ctx, envOverrides)
 	return ctx, nil
 }
 
+func (r *runnerService) getGitToken(ctx context.Context, groupID string, repoURL string) (string, error) {
```

Comment:
nit: avoid passing in the group ID here (best to avoid plumbing around group IDs as raw strings as much as we can, since strings don't inherently encode any auth info)

if you call `u.GetGroupID()` inside this func then you can also skip calling AuthorizeGroupAccess here because `GetGroupID` always returns an authorized group ID